src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -2198,17 +2201,7 @@ public override SortedSet<T> GetViewBetween(T lowerValue, T upperValue)
 
             internal override void IntersectWithEnumerable(IEnumerable<T> other)
             {
-                List<T> toSave = new List<T>(this.Count);
-                foreach (T item in other)
-                {
-                    if (Contains(item))
-                    {
-                        toSave.Add(item);
-                        Remove(item);
-                    }
-                }
-                Clear();
-                AddAllElements(toSave);
+                base.IntersectWithEnumerable(other);
```

Comment:
Went with the latter. The same debug statement appears in several other places, so it seems worthwhile to keep.